### PR TITLE
REF/TYP: pandas/core/window/*.py

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

* Create clearer separation for methods belonging to `Window` vs `Rolling`
* More typing

### CODE REVIEW

File: pandas/core/window/expanding.py [hunk 1]
```diff
@@ -245,12 +250,23 @@ def quantile(self, quantile, interpolation="linear", **kwargs):
     @Substitution(name="expanding", func_name="cov")
     @Appender(_doc_template)
     @Appender(_shared_docs["cov"])
-    def cov(self, other=None, pairwise=None, ddof=1, **kwargs):
+    def cov(
+        self,
+        other: Optional[Union[np.ndarray, FrameOrSeries]] = None,
```
Comment (jreback) [inline pandas/core/window/expanding.py:255]: do we actually allow other to be a non-frame/series?

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, cc @simonjayhawkins

File: pandas/core/window/expanding.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/window/expanding.py:255]: Yeah we have tests where `other` is an ndarray

File: pandas/core/window/expanding.py [hunk 2]
```diff
@@ -135,12 +140,12 @@ def count(self, **kwargs):
...
         self,
-        func,
+        func: Callable,
         raw: bool = False,
         engine: Optional[str] = None,
         engine_kwargs: Optional[Dict[str, bool]] = None,
-        args=None,
-        kwargs=None,
+        args: Optional[Tuple] = None,
+        kwargs: Optional[Dict] = None,
```
Comment (simonjayhawkins) [inline pandas/core/window/expanding.py:148]: we will eventually want to use disallow_any_generic as we increase strictness checks xref #30539

so for kwargs could probably use `Dict[str, Any]` and for args, `Tuple[Any, ...]`

can you also add type parameters to Callable.

Comment (mroeschke) [reply pandas/core/window/expanding.py]: Sounds good. Done

Comment (simonjayhawkins) [reply pandas/core/window/expanding.py]: Thanks @mroeschke lgtm

**Review by jreback — APPROVED** [review]

File: pandas/core/window/ewm.py [hunk 3]
```diff
@@ -60,6 +64,15 @@ def get_center_of_mass(
     return float(comass)
 
 
+def wrap_result(obj: "Series", result: np.ndarray) -> "Series":
```
Comment (jreback) [inline pandas/core/window/ewm.py:67]: for clarify should make this `wrap_result_1d`, but can be in the future

**Pull request merged**